commands/eval.md

```diff
@@ -194,6 +194,24 @@ returned in the format specified above (as a Lua table with an `err` field).
 The script can pass the exact error to the user by returning the error object
 returned by `redis.pcall()`.
 
+## Running Lua under low memory conditions
+
+When the memory usage in Redis exceeds the `maxmemory` limit, the first write command
+encountered in the Lua script that uses additional memory will cause the script to
+abort. However, one thing to caution here is that if the first write command does
```

Comment:
technically, it'll cause the script to abort if `redis.call` was used, but not if `redis.pcall` was used.
```suggestion
encountered in the Lua script that uses additional memory will fail with -OOM error.
However, one thing to caution here is that if the first write command does
```

Reply:
I think Redis will abort the script regardless of whether it is `lua.call` or `lua.pcall`. See https://github.com/redis/redis/blob/unstable/src/scripting.c#L663-L676

Reply:
Right. Sorry, I mixed it up with commands that return an error.

Reply:
@oranagra I think you are correct, just tested, if redis.pcall is used the script is not aborted:
```
127.0.0.1:6379> set x 1
(error) OOM command not allowed when used memory > 'maxmemory'.
127.0.0.1:6379> EVAL "redis.call('set', 'x', '1'); return 1" 0
(error) ERR Error running script (call to f_05c76db2bc9e1d0e1c646a7959c5d6ee8177c884): @user_script:1: @user_script: 1: -OOM command not allowed when used memory > 'maxmemory'.
127.0.0.1:6379> EVAL "redis.pcall('set', 'x', '1'); return 1" 0
(integer) 1
```

Reply:
Thanks.. Good to know. (assuming I'll remember)

Reply:
But its not only that, the script might also decide to catch the error raised by redis.call so even on redis.call it might not be aborted:
```
127.0.0.1:6379> EVAL "local f = function() redis.call('set', 'x', '1'); end; f() ; return 1" 0
(error) ERR Error running script (call to f_de1355275acdfc1349db5a1a34a0f5db57ff2642): @user_script:1: @user_script: 1: -OOM command not allowed when used memory > 'maxmemory'.
127.0.0.1:6379> EVAL "local f = function() redis.call('set', 'x', '1'); end; pcall(f); return 1" 0
(integer) 1
```

Reply:
Yes, that's actually all very logical, it was my initial intuition (so no need to remember). 
I was just too busy / lazy to check.

Reply:
I just ran the same Lua test. I noticed that when you do `redis.pcall()`, the script actually aborts internally without generating an error response back to the client. So it looks like the client command succeeded, but the actual write fails in Redis silently. 

```
127.0.0.1:6379> set a 1
(error) OOM command not allowed when used memory > 'maxmemory'.
127.0.0.1:6379> EVAL "redis.call('set', 'x', '1'); return 1" 0
(error) ERR Error running script (call to f_05c76db2bc9e1d0e1c646a7959c5d6ee8177c884): @user_script:1: @user_script: 1: -OOM command not allowed when used memory > 'maxmemory'.
127.0.0.1:6379> scan 0
1) "0"
2) (empty array)
127.0.0.1:6379> EVAL "redis.pcall('set', 'x', '1'); return 1" 0
(integer) 1
127.0.0.1:6379> scan 0
1) "0"
2) (empty array)
```

Reply:
Yes, pcall will not failed your script. If you check the return value of pcall you will see the error and you have to handle the error in your script.

Reply:
Do you think it is a good idea if I add another note here that Redis silently fails the write if you are not catching the error when using `lua.pcall()`? I feel that is already a general functionality of `lua.pcall()` that it will not fail your script unless you handle it inside your script.

Reply:
@QuChen88 i suppose more details are always better, but considering the section about the difference between `call` and `pcall` is right above this one, i guess it should be short, maybe something like:
```diff
- will cause the script to abort
+ will cause the script to abort (unless `redis.pcall` was used)
```
this way it's just a tip, sending the reader to get more details in the other section.

also, looking again at the last sentence:
> In those scenarios, it is recommended to configure the `maxmemory-policy` appropriately to allow the proper evictions of items to free up memory in between Lua scripts.

i think it's still unclear what you mean to say, i suggest extending it a bit, maybe give an example of the danger and how to avoid it.

Reply:
Fixed.